mod/wikirate_source/spec/set/type_plus_right/source/file-import_spec.rb

```diff
@@ -0,0 +1,88 @@
+describe Card::Set::TypePlusRight::Source::File::Import do
+  before do
+    login_as 'joe_user' 
+    test_csv = File.new "#{Rails.root}/mod/wikirate_source/spec/set/type_plus_right/source/import_test.csv"
+    file_uploaded = ActionDispatch::Http::UploadedFile.new(:tempfile => test_csv, :filename => File.basename(test_csv))
+    
+    @source = Card.create! :type_id=>Card::SourceID,:subcards=>{'+File'=>{ :attach=>file_uploaded,:content=>"CHOSEN",:type_id=>Card::FileID}}
+  end
+  describe "while adding metric value" do
+    it "shows errors while params do not fit" do 
+
+      Card::Env.params["is_metric_import_update"] = 'true'
+      source_file = @source.fetch :trait=>:file
+      source_file.update_attributes :subcards=>{"#{@source.name}+#{Card[:metric].name}"=>{:content=>'[[Access to Nutrition Index+Marketing Score]]',:type_id=>Card::PointerID}}
+
+      
+      expect(source_file.errors).to have_key(:content)
+      expect(source_file.errors[:content]).to include("Please give a year.")
+      
+      source_file.update_attributes :subcards=>{"#{@source.name}+#{Card[:year].name}"=>{:content=>'[[2015]]',:type_id=>Card::PointerID}}
+      
+      expect(source_file.errors).to have_key(:content)
+      expect(source_file.errors[:content]).to include("Please give a metric.")
+
+    end
+    it "adds correct metric values" do 
+      
+      Card::Env.params["is_metric_import_update"] = 'true'
+      Card::Env.params[:metric_values] = {"Amazon"=>["9"], "Apple Inc."=>["62"]}
+      source_file = @source.fetch :trait=>:file
+      source_file.update_attributes :subcards=>{"#{@source.name}+#{Card[:metric].name}"=>{:content=>'[[Access to Nutrition Index+Marketing Score]]',:type_id=>Card::PointerID},"#{@source.name}+#{Card[:year].name}"=>{:content=>'[[2015]]',:type_id=>Card::PointerID}}
+
+      expect(Card.exists?("Access to Nutrition Index+Marketing Score+Amazon+2015")).to be true
+      expect(Card.exists?("Access to Nutrition Index+Marketing Score+Apple Inc.+2015")).to be true
+
+      amazon_2015_metric_value_card = Card["Access to Nutrition Index+Marketing Score+Amazon+2015+value"]
+      apple_2015_metric_value_card = Card["Access to Nutrition Index+Marketing Score+Apple Inc.+2015+value"]
+      expect(amazon_2015_metric_value_card.content).to eq("9")
+      expect(apple_2015_metric_value_card.content).to eq("62")
+      
+
+    end
+  end
+  describe "while rendering import view" do
+    
+    it "shows field correctly" do
+      
+      source_file_card = @source.fetch :trait=>:file
+      html = source_file_card.format.render_import      
+
```

Comment:
I think I would be tempted to do binding.pry, run this both individually and with all specs, and then see what you can do to find something different in the two.  check Card::Env, Card::Auth, the local cache, etc.  Just be careful not to change much in the whole-spec environment (so hard to get back there again!)